pytorch_lightning/models/trainer.py

```diff
@@ -356,12 +356,10 @@ def __layout_bookeeping(self):
 
         # determine number of validation batches
         # val datasets could be none, 1 or 2+
-        self.nb_val_batches = 0
         if self.val_dataloader is not None:
```

Comment:
is it still needed since you drop None?
I would replace by asserting that `self.val_dataloader is not None`